### ISSUE TITLE
multiple pre-assignment of tokens

### ISSUE THREAD

**Issue by ethers** [ConsenSys/gnosis-contracts#29]
Found by Gonçalo Sá @GNSPS

This is a minor because `owners` getting a larger pre-assignment of Gnosis tokens without being detected is highly unlikely for an auction that is likely to be highly visible and scrutinized.  However, this type of issue is currently subtle and important to raise awareness.  For example, less scrutinized auctions may use such a technique to defraud the community.  Or this vector might be able to be combined with other deliberate underhanded coding techniques.  The proposed solution below is simple and removes doubt from the code: it should also be documented explicitly as preventing multiple pre-assignment (as otherwise it's easy to just interpret the solution as "null checks" [which are generally good practice but misses the key point in this case]).

TLDR: By setting these addresses to zero (`address _gnosisToken, address _wallet`), invoking `setup` multiple times allows `owners` to get more tokens (before the auction starts, thus is minor).

-------
Gonçalo's report and recommendation verbatim:
#### <a id="heading-4.1.4.1"/> 4.1.4.1 Unclear condition for locking the `setup()` function [*Minor* Issue]

The `setup()` function is meant to be ran only once (by design, as seen in the comment below) and to enforce that it has the following condition before any state changes are performed:

```
if (wallet != 0 || address(gnosisToken) != 0) {
    // Setup was executed already
    throw;
}
```

The issue that arises is related to the use of the disjunction (**||**) in the condition. **Is any of these variables equal to *0* in any possibly wanted execution scenario?** Practically speaking, with the use of the *OR* condition it becomes possible for one of the variables to be set as null.

As this function is run for setting up the auction, in a pre-public-launch fashion, any error would be caught before any harm was done. However, once again following smart contracts' design best practices, it is always good to have those restrictions explicitly set in the code and, in this case, there's no single scenario in which having the `wallet` variable *null* is right (or `gnosisToken`, for that matter).

Also, this issue could lead to a different problem: **multiple pre-assignment of tokens** if the arguments were sent both *null*ed out and the `owners` and `tokens` arrays were correctly set.

**Recommended Solution**:

Change the condition to check for *null* arguments:

```
if (wallet != 0 || address(gnosisToken) != 0 || _wallet == 0 || _gnosisToken == 0) {
    // Setup was executed already
    // or arguments are null
    throw;
}
```

**Comment by Georgi87** [comment]
I have added now validation for function arguments for all constructor and setup functions: https://github.com/ConsenSys/gnosis-contracts/commit/ea1b2bd00587c8ce32e49b765e2b161bc5041e35
